pipeline-maven/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/JacocoReportPublisher.java

```diff
@@ -123,184 +112,11 @@ public JacocoReportPublisher() {}
-        }
-
-        JacocoPublisher jacocoPublisher = new JacocoPublisher();
-
-        String aggregatedDestFile =
-                jacocoReportDetails.stream().map(details -> details.execFile).collect(Collectors.joining(","));
-        String aggregatedSourceDirectory = jacocoReportDetails.stream()
-                .map(details -> details.sourceDirectory)
-                .collect(Collectors.joining(","));
-        String aggregatedClassesDirectory = jacocoReportDetails.stream()
-                .map(details -> details.classesDirectory)
-                .collect(Collectors.joining(","));
-        String aggregatedExclusionPattern = jacocoReportDetails.stream()
-                .map(details -> details.excludes)
-                .filter(e -> e != null && !e.isEmpty())
-                .distinct()
-                .collect(Collectors.joining(","));
-
-        jacocoPublisher.setExecPattern(aggregatedDestFile);
-        jacocoPublisher.setSourcePattern(aggregatedSourceDirectory);
-        jacocoPublisher.setClassPattern(aggregatedClassesDirectory);
-        jacocoPublisher.setExclusionPattern(aggregatedExclusionPattern);
-
-        try {
-            jacocoPublisher.perform(run, workspace, launcher, listener);
-        } catch (Exception e) {
-            listener.error("[withMaven] jacocoPublisher - exception archiving JaCoCo results for " + jacocoReportDetails
-                    + ": " + e);
-            LOGGER.log(Level.WARNING, "Exception processing JaCoCo results", e);
-            throw new MavenPipelinePublisherException(
-                    "jacocoPublisher", "archiving JaCoCo results for " + jacocoReportDetails, e);
-        }
-    }
-
-    public static class JacocoReportDetails {
-        final String execFile, sourceDirectory, classesDirectory, excludes, description;
-
-        public JacocoReportDetails(
-                String execFile, String sourceDirectory, String classesDirectory, String excludes, String description) {
-            this.execFile = execFile;
-            this.sourceDirectory = sourceDirectory;
-            this.classesDirectory = classesDirectory;
-            this.excludes = excludes;
-            this.description = description;
-        }
-
-        @Override
-        public String toString() {
-            return "JacocoReportDetails{" + "execFile='"
-                    + execFile + '\'' + ", sourceDirectory='"
-                    + sourceDirectory + '\'' + ", classesDirectory='"
-                    + classesDirectory + '\'' + ", excludes='"
-                    + excludes + '\'' + ", description='"
-                    + description + '\'' + '}';
-        }
+        throw new AbortException(
+                """
+        The jacocoPublisher is deprecated as is the Jacoco plugin and you should not use it.
+        Alternatively, you should rely on Coverage plugin and see the configuration required at https://plugins.jenkins.io/coverage/#plugin-content-pipeline-example.
```

Comment:
Or, if continuing to use the JaCoCo plugin for now, run it explicitly rather than expecting `withMaven` to run it implicitly.